mesonbuild/interpreter.py

```diff
@@ -1798,7 +1798,12 @@ def dependency_fallback(self, name, kwargs):
         dirname, varname = self.get_subproject_infos(kwargs)
         # Try to execute the subproject
         try:
-            self.do_subproject(dirname, {})
+            sp_kwargs = {}
+            try:
+                sp_kwargs['default_options'] = kwargs['default_options']
```

Comment:
`try ... except` is actually the slowest possible way to do this. Even slower than `.get()`. This is the fastest way:

```
if 'key' in kwargs:
  sp_kwargs[key] = kwargs[key]
```